CMakeLists.txt

```diff
@@ -3,20 +3,49 @@ cmake_minimum_required(VERSION 2.6.4)
 # http://www.cmake.org/Bug/view.php?id=8598
 # http://gcc.gnu.org/bugzilla/show_bug.cgi?id=35707
 # http://www.cmake.org/Bug/view.php?id=8408
-project(PARTMC Fortran C)
+if (ENABLE_TCHEM)
+  # Use the wrapper to nvcc for the CXX compiler for GPU
+  if (ENABLE_TCHEM_GPU)
+    set(ENV{CXX} $ENV{KOKKOS_PATH}/bin/nvcc_wrapper)
+    project(PARTMC Fortran C CXX)
+  else()
+    project(PARTMC Fortran C)
+  endif()
+else()
+  project(PARTMC Fortran C)
+endif()
 
 set(PACKAGE_BUGREPORT "mwest@illinois.edu")
 set(PACKAGE_NAME "PartMC")
 set(PACKAGE_STRING "PartMC 2.8.0")
 set(PACKAGE_TARNAME "partmc")
 set(PACKAGE_VERSION "2.8.0")
 
+include(FortranCInterface)
+
+# Following must be commented out if using -Wimplicit-interface with -Werror
+# Working with C and Fortran
+#FortranCInterface_VERIFY()
+#if (ENABLE_TCHEM_GPU)
+#   # Working with C++ and Fortran
+#   FortranCInterface_VERIFY(CXX)
+#endif()
```

Comment:
Any idea why this causes problems?

Reply:
The code for verifying Fortran/C/C++ compatibility is using an implicit interface. This triggers a warning and we error out on all warnings with the build flags used in CI. And then the check fails (and therefore the whole process fails). Example code from CMake (Modules/FortranCInterface/main.F)

```
        program main
          call call_sub()
#ifdef CALL_MOD
          call call_mod()
#endif
        end
```

Reply:
Ah ok, that's annoying but I guess we are stuck with it.